File: API/Entities/OrderAggregates/Order.cs
namespace API.Entities.OrderAggregates
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        public ShippingAdress ShippingAdress { get; set; }
        public DateTime OrderDate = DateTime.Now;
        public List<OrederItem>  OrederItems{ get; set; }
        public long SubTotal { get; set; }
        public long DeleveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; }=OrderStatus.Pending;
        public long GetTotal()
        {
            return SubTotal + DeleveryFee;
        }
    }
}

File: client/src/App/Models/Order.ts
export interface Order {
  id: number;
  buyerId: string;
  shippingAdress: ShippingAdress;
  orderDate: string;
  orederItems: OrederItem[];
  subTotal: number;
  deleveryFee: number;
  orderStatus: string;
  total: number;
}

export interface ShippingAdress {
  fullName: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  zip: string;
  country: string;
}

export interface OrederItem {
  productId: number;
  name: string;
  pictureURL: string;
  price: number;
  quantity: number;
}
